common/distribution/dist.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distribution
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// DistType represents a distribution type
+type DistType string
+
+const (
+	DistScheme = "cimd"
+
+	// Distribution types
+	DistTypeAppc       DistType = "appc"
```

Comment:
I was thinking that forcing a Distribution to implement `ComparableString() string` will add two things in one:

* Have a ready to use and reproducible distribution serialization string.
* avoid them implementing a Compare method and use the common Compare function.

But these can also be done removing the `ComparableString() string` method from the interface and add a `Compare(Distribution) bool` method. For distribution serialization just URI().String() could be used (so an user isn't tempted to just compare two distribution string but use the `Compare` method.
It'll probably be cleaner. Will experiment with this.